ydb/core/kqp/gateway/kqp_ic_gateway.cpp

```diff
@@ -899,6 +896,47 @@ class TKikimrIcGateway : public IKqpGateway {
         return tablePromise.GetFuture();
     }
 
+    TFuture<TGenericResult> AlterDatabase(const TString& cluster, const NYql::TAlterDatabaseSettings& settings) override {
+        using TRequest = TEvTxUserProxy::TEvProposeTransaction;
+
+        try {
+            if (!CheckCluster(cluster)) {
+                return InvalidCluster<TGenericResult>(cluster);
+            }
+
+            auto alterDatabasePromise = NewPromise<TGenericResult>();
+
+            auto ev = MakeHolder<TRequest>();
+
+            if (UserToken) {
```

Comment:
Please also set `DatabaseName` field (`ev->Record.SetDatabaseName()`).
Otherwise, requests for tenant databases will be send to and executed at the root database. Which should work but that's not how other modify-acl requests are working out.